scw/errors.go

```diff
@@ -185,34 +195,62 @@ func (e *InvalidArgumentsError) GetRawBody() json.RawMessage {
 	return e.RawBody
 }
 
-// InvalidRequestError are only returned by the compute API
+// InvalidRequestError is only returned by the instance API.
 // Warning: this is not a standard error.
 type InvalidRequestError struct {
+	Message string `json:"message"`
+
 	Fields map[string][]string `json:"fields"`
 
 	RawBody json.RawMessage `json:"-"`
 }
 
-// ToInvalidArgumentsError converts it to the standard error InvalidArgumentsError
-func (e *InvalidRequestError) ToInvalidArgumentsError() *InvalidArgumentsError {
-	invalidArguments := &InvalidArgumentsError{
-		RawBody: e.RawBody,
+// ToSdkError converts it to the standard error InvalidArgumentsError
+func (e *InvalidRequestError) ToSdkError() SdkError {
+	// If error has fields, it is an invalid arguments error.
+	if e.Fields != nil {
+		invalidArguments := &InvalidArgumentsError{
+			RawBody: e.RawBody,
+		}
+		fieldNames := []string(nil)
+		for fieldName := range e.Fields {
+			fieldNames = append(fieldNames, fieldName)
+		}
+		sort.Strings(fieldNames)
+		for _, fieldName := range fieldNames {
+			for _, message := range e.Fields[fieldName] {
+				invalidArguments.Details = append(invalidArguments.Details, InvalidArgumentsErrorDetail{
+					ArgumentName: fieldName,
+					Reason:       "constraint",
+					HelpMessage:  message,
+				})
+			}
+		}
+		return invalidArguments
 	}
-	fieldNames := []string(nil)
-	for fieldName := range e.Fields {
-		fieldNames = append(fieldNames, fieldName)
+
+	return &ResponseError{
+		Message: strings.ToLower(e.Message),
+		Type:    "invalid_request_error",
+		RawBody: e.RawBody,
 	}
-	sort.Strings(fieldNames)
-	for _, fieldName := range fieldNames {
-		for _, message := range e.Fields[fieldName] {
-			invalidArguments.Details = append(invalidArguments.Details, InvalidArgumentsErrorDetail{
-				ArgumentName: fieldName,
-				Reason:       "constraint",
-				HelpMessage:  message,
-			})
-		}
+}
+
+// ConflictError is only returned by the instance API.
+// Warning: this is not a standard error.
+type ConflictError struct {
+	Message string `json:"message"`
+
+	RawBody json.RawMessage `json:"-"`
+}
+
+// ToSdkError converts it to the standard error InvalidArgumentsError
+func (e *ConflictError) ToSdkError() SdkError {
+	return &ResponseError{
```

Comment:
ToSdkError should return a standard error. The whole point of `unmarshalNonStandardError` is to convert the body ==> into a non-standard error ==> into a "faked" standard error. By default every error not catched by `unmarshalNonStandardError` and `unmarshalStandardError` will already return a `ResponseError`.